python/paddle/fluid/layers/nn.py

```diff
@@ -3346,6 +3347,94 @@ def group_norm(input,
     return helper.append_activation(group_norm_out)
 
 
+@templatedoc()
+def spectral_norm(weight, dim=0, power_iters=1, eps=1e-12, name=None):
+    """
+    **Spectral Normalization Layer**
+
+    This layer calculate the spectral normalize value of weight parameters of
+    fc, conv1d, conv2d, conv3d layers which should be 2-D, 3-D, 4-D, 5-D
+    Parameters. Calculations are showed as followings.
+
+    .. code-block:: text
+
+        Step 1:
+        Generate vector u in shape of [h], and v in shape of [w].
+        While h is the attr:`dim`th dimension of the input weights,
+        and w is the product result of remain dimensions.
+
+        Step 2:
+        While attr:`power_iters` is a positive interger, do following
+        iteration calculations with u and v for attr:`power_iters`
+        round.
+            \mathbf{v} = \mathbf{W}^{T} \mathbf{u}
+            \mathbf{v} = \frac{\mathbf{v}}{\|\mathbf{v}\|_2}
+            \mathbf{u} = \mathbf{W}^{T} \mathbf{v}
+            \mathbf{u} = \frac{\mathbf{u}}{\|\mathbf{u}\|_2}
+
+        Step 3:
+        Calculate \sigma{W} and scale weight values.
```

Comment:
```suggestion
        Calculate \sigma{W} and normalize weight values.
```
'scale' may cause confusion. I think "normalize" is closer to our context

Reply:
Done